src/types/routePaths.ts

```diff
@@ -0,0 +1,55 @@
+import { ExtractRouteMethodParams } from '@/types/routeMethods'
+import { Public, Route, Routes } from '@/types/routes'
+
+export type ExtractRoutePathParameters<
+  TRouteMethods,
+  TRoutePath extends string
+> = TRoutePath extends string ? NestedObjectValue<TRouteMethods, TRoutePath> : never
+
+type NestedObjectValue<
+  TRouteMethods,
+  TRoutePath extends string
+> =
+  TRoutePath extends `${infer F}.${infer R}`
+    ? F extends keyof TRouteMethods
+      ? NestedObjectValue<TRouteMethods[F], R>
+      : never
+    : TRoutePath extends keyof TRouteMethods
+      ? ExtractRouteMethodParams<TRouteMethods[TRoutePath]>
+      : never
+
+export type RoutePaths<
+  TRoutes extends Routes,
+  TPrefix extends string = ''
+> = FlatArray<{
+  [K in keyof TRoutes]: TRoutes[K] extends Route
+    ? [RoutePath<TRoutes[K], TPrefix>, ChildrenPaths<TRoutes[K], TPrefix>]
+    : never
+}, 1>
+
+type RoutePath<
+  TRoute extends Route,
+  TPrefix extends string
+> = TRoute extends Public<TRoute> & { name: infer Name extends string }
```

Comment:
public just means it's callable yeah? I think we just need a name for a parent to show up in RoutePath

Reply:
Yeah this is incorrect. Removed the `Public<TRoute>` check.